app/lib/core/services/service_locator.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:app/core/view_models/nav_view_model.dart';
+import 'package:app/core/view_models/signin_view_model.dart';
+import 'package:app/core/view_models/signout_view_model.dart';
+import 'package:app/core/view_models/welcome_view_model.dart';
+import 'package:get_it/get_it.dart';
+import 'authentication/auth_service.dart';
+import 'navigation/nav_service.dart';
+
+final locator = GetIt.instance;
+
+void setUpLocator() {
+  locator.registerLazySingleton<AuthenticationService>(
```

Comment:
It makes sense to have services as singletons, but not widget factories as a singleton registered in `GetIt`. At least, to me.

Reply:
trust me, this is needed. It wont work otherwise